litex/soc/integration/common.py

```diff
@@ -80,11 +80,11 @@ def get_mem_data(filename_or_regions, data_width=32, endianness="big", mem_size=
                     "little": "<I",
                     "big":    ">I"
                 }[endianness]
-                if data_width == 32:
-                    data[(base - offset)//bytes_per_data + i] = struct.unpack(unpack_order, w)[0]
-                if data_width == 64:
-                    data[(base - offset)//bytes_per_data + i]  = (struct.unpack(unpack_order, w[0:4])[0] <<  0)
-                    data[(base - offset)//bytes_per_data + i] |= (struct.unpack(unpack_order, w[4:8])[0] << 32)
+                data[(base - offset)//bytes_per_data + i] = 0
+                for filled_data_width in range(0, data_width, 32):
+                    cur_byte = filled_data_width//8
+                    data[(base - offset)//bytes_per_data + i] |= (struct.unpack(unpack_order, w[cur_byte:cur_byte+4])[0] << filled_data_width)
+                    filled_data_width += 32
```

Comment:
```suggestion
```

Hmm, this should be a line of dead code as the for-loop will always set it to be the right value.

Reply:
Indeed, thanks @Rongronggg9, this is fixed with https://github.com/enjoy-digital/litex/commit/8f459a27dda79789094ce400a885ecf2a902a768.